torch/csrc/jit/codegen/cuda/kernel_db/kernel_db.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+
+#if defined(__cplusplus) && (__cplusplus >= 201703L)
+#include <filesystem>
+namespace fs = std::filesystem;
+#elif defined(__cplusplus) && (__cplusplus >= 201402L)
+#include <experimental/filesystem>
+namespace fs = std::experimental::filesystem;
+#else
+#error "C++14 or Higher is required for filesystem library!"
+#endif
+
+#include <unordered_map>
+#include <vector>
+
+#include <c10/macros/Export.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+//! KernelDbEntry captures information to be printed per fusion in a csv file
+//! that is used to restore a hash map
+struct KernelDbEntry {
+  //! Cuda kernel function signature that is required to load the Cubin
+  std::string kernel_name;
```

Comment:
So this isn't just the name but the entire kernel signature?

Reply:
It is the kernel signature.  I think that seems to suggest a name change. :-)   It is used to load a module.